doubly_linked_list/doubly_linked_list.py

```diff
@@ -44,25 +44,102 @@ def __len__(self):
     return self.length
 
   def add_to_head(self, value):
-    pass
-
+    self.length +=1
+    new_node = ListNode(value = value)
+    new_node.next = None
+    new_node.prev = None
+    if self.head is None:
+      self.head = new_node
+      self.tail = new_node
+    else:
+      self.head.prev = new_node
+      new_node.next = self.head
+      self.head = new_node
+    
   def remove_from_head(self):
-    pass
+    if self.head == None:
+      return
+    else:
+      self.length -=1
+      if self.head != self.tail:
+        self.head = self.head.next
+        self.head.prev = None
+      else:
+        self.head = self.tail=None
```

Comment:
You're having an issue here because you need to be removing the node and returning the value of the removed node. I used the code:

if not self.head and not self.tail:
____return None
self.length -= 1
if self.head == self.tail:
____current_head = self.head
____self.head = None
____self.tail = None
____return current_head.value
current_head = self.head
self.head = self.head.next
self.head.prev = None
return current_head.value